src/Sparrow/ByteString.cs

```diff
@@ -755,7 +730,12 @@ public ByteStringContext(SharedMultipleUseFlag lowMemoryFlag, int allocationBloc
         public void Reset()
         {
             if (_disposed)
+            {
+                if (_isFinalizerThread)
```

Comment:
Can this now _ever_ be called from finalizer?

Reply:
no, we can remove _isFinalizerThread entirely